core/vault/src/main/java/uk/gov/dwp/vault/SingleValueVaultLookupStrategy.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.dwp.vault;
+
+
+import com.bettercloud.vault.Vault;
+import com.bettercloud.vault.VaultException;
+
+import org.springframework.util.Assert;
+
+import uk.gov.dwp.vault.config.VaultProperties;
+import uk.gov.dwp.vault.domain.DecryptedValue;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This assumes that values looked up in vault are basically in the format of "<VALUE_KEY>=<SECRET_VALUE>". Vault is capable of storing an object as the secret, but this strategy is only concerned
+ * with single field values defined by the key called "value"
+ */
+public class SingleValueVaultLookupStrategy implements SensitiveConfigValueLookupStrategy {
+
+    private static final String VALUE_KEY = "value";
+    private static final Pattern PATTERN = Pattern.compile("VAULT\\((.+)\\)");
+
+    private final VaultProperties vaultProperties;
+    private final VaultApiFactory vaultApiFactory;
+
+
+    public SingleValueVaultLookupStrategy(VaultProperties vaultProperties, VaultApiFactory vaultApiFactory) {
+        this.vaultProperties = vaultProperties;
+        this.vaultApiFactory = vaultApiFactory;
+    }
+
+    @Override
+    public boolean matches(String secret) {
+        return isVaultEnabledAndPatternMatches(secret);
+    }
+
+    @Override
+    public int evaluationOrder() {
+        return FIRST_EVAL_ORDER;
+    }
+
+    @Override
+    public DecryptedValue retrieveSecret(String secretPath) {
+        Matcher matcher = PATTERN.matcher(secretPath);
+        if (matcher.matches()) {
+            String vaultPath = matcher.group(1);
+            try {
+                String value = vaultAPI().logical().read(vaultPath).getData().get(VALUE_KEY);
```

Comment:
The VaultAPI seems floored.  The `read(vaultPath)`  method returns a `LogicalResponse` which stores all your secrets in a `HashMap<String, String>()` .  Meaning all the "secrets" are now stored in the Java String Pool.  From Java 7 Strings are stored in Heap space (rather than PermGen), which is GC'd.

Perhaps we should in-line this code so that we're not referencing yet another copy of the secret as a String